indexing-service/src/main/java/org/apache/druid/indexing/overlord/ForkingTaskRunner.java

```diff
@@ -316,7 +317,7 @@ public TaskStatus call()
                             if (propName.startsWith(CHILD_PROPERTY_PREFIX)) {
                               command.addSystemProperty(
                                   propName.substring(CHILD_PROPERTY_PREFIX.length()),
-                                  task.getContextValue(propName)
+                                  Preconditions.checkNotNull(task.getContextValue(propName), propName).toString()
```

Comment:
I don't think we should throw an exception. Maybe a property was set to `null` on purpose. We should just use the value of null.

Maybe something like this:
```java
if (propName.startsWith(CHILD_PROPERTY_PREFIX)) {
    final String contextValue = task.getContextValue(propName);
    command.addSystemProperty(
        propName.substring(CHILD_PROPERTY_PREFIX.length()),
        contextValue == null ? null : contextValue.toString()
    );
}    
```